src/manimVersion.ts

```diff
@@ -1,22 +1,31 @@
-import * as vscode from "vscode";
-import { window } from "vscode";
-import { waitNewTerminalDelay, onTerminalOutput } from "./utils/terminal";
+import { exec } from "child_process";
+
 import { Window, Logger } from "./logger";
-import { ManimShell } from "./manimShell";
-import { manimNotebookContext } from "./extension";
+import * as vscode from "vscode";
 
 /**
  * Manim version that the user has installed without the 'v' prefix,
  * e.g. '1.2.3'.
  */
 let MANIM_VERSION: string | undefined;
-let isCanceledByUser = false;
 
 /**
- * Key for the global state to store the last warning time for missing Manim
- * version detection.
+ * The last Python binary path that was used to determine the Manim version.
+ *
+ * This is used to retry the version determination if it failed the first time.
+ */
+let lastPythonBinary: string | undefined;
+
+/**
+ * The URL to the Manim releases page.
+ */
+const MANIM_RELEASES_URL = "https://github.com/3b1b/manim/releases";
+
+/**
+ * The URL to the API endpoint showing information about the latest
+ * Manim release.
  */
-export const LAST_WARNING_NO_VERSION_KEY = "manim-notebook.lastWarningTimeMissingVersionDetection";
+const MANIM_LATEST_RELEASE_API_URL = "https://api.github.com/repos/3b1b/manim/releases/latest";
```

Comment:
Lol, I was just wondering why I'm getting `You're using ManimGL version ...` instead of the expected `You're using the latest ManimGL version ...`. It's because this API endpoint was just rate-limited for my IP 🙈 Might be the case for other users as well, but at least they have a chance to find out about the latest Manim version available.